src/cart/crt_init.c

```diff
@@ -199,6 +199,14 @@ prov_data_init(struct crt_prov_gdata *prov_data, crt_provider_t provider, bool p
 	prov_data->cpg_max_unexp_size = max_unexpect_size;
 	prov_data->cpg_primary        = primary;
 
+	if (opt && opt->cio_progress_busy) {
+		prov_data->cpg_progress_busy = opt->cio_progress_busy;
+	} else {
+		bool progress_busy = false;
+		crt_env_get(D_PROGRESS_BUSY, &progress_busy);
+		prov_data->cpg_progress_busy = progress_busy;
+	}
+
```

Comment:
minor: should we emit a warning when cio_progress_busy or envar are set? The note says its intended for testing purposes only, but feels can be easily missed if someone sets it and forgets

Reply:
we could but that could also lead to a lot of output if we have a lot of processes, i.e. I would not necessarily want it to be a warning if we're testing something. I'm okay to add an info message though.

Reply:
i dont think info message is needed tbh, as we already dump both envs and cio flags at init at info level. oh... speaking of which, you do need to update the dump init opt routine :)

Reply:
see line 75

Reply:
ah ok thanks just updated.